File: 1957/Solution.go
func makeFancyString(s string) string {
    var prev rune
    var res strings.Builder
    freq := 0

    for _, c := range s {
        if c == prev {
            freq++
        } else {
            freq = 1
            prev = c
        }
        if freq < 3 {
            res.WriteRune(c)
        }
    }

    return res.String()
}

File: 1957/Solution.rs
impl Solution {
    pub fn make_fancy_string(s: String) -> String {
        let mut res = String::new();
        let mut prev = '\0';
        let mut freq = 0;
        for c in s.chars() {
            if c == prev {
                freq += 1;
            } else {
                freq = 1;
                prev = c;
            }
            if freq < 3 {
                res.push(c);
            }
        }
        res
    }
}
